go_modules/lib/dependabot/go_modules/file_updater.rb

```diff
@@ -67,7 +67,10 @@ def use_repo_contents_stub
         @repo_contents_path = Dir.mktmpdir
         Dir.chdir(@repo_contents_path) do
           dependency_files.each do |file|
-            File.write(file.name, file.content)
+            path = File.join(@repo_contents_path, directory, file.name)
+            path = Pathname.new(path).expand_path
+            FileUtils.mkdir_p(path.dirname) unless Dir.exist?(path.dirname)
+            File.write(path, file.content)
```

Comment:
Not sure this is used but was causing problems without the dry-run clone changes

Reply:
It should not be used in prod, but people might not perform full clones when running standalone

Reply:
Added a test case for this and fixed the global git config messing with my local config outside of docker, accidentally committed with `no-reply@github.com` 😅